lib/kubernetes-deploy/resource_discovery.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module KubernetesDeploy
+  class ResourceDiscovery
+    def initialize(namespace:, context:, logger:, namespace_tags:)
+      @namespace = namespace
+      @context = context
+      @logger = logger
+      @namespace_tags = namespace_tags
+      @cr_mapping = {}
+    end
+
+    def crds(sync_mediator)
+      sync_mediator.get_all(CustomResourceDefinition.kind).map do |r_def|
```

Comment:
Since we are not walking the entire API tree we can only express dependencies between resources that are explicitly modelled in KubernetesDeploy (i.e. there is a class defined for them in this project) or CRDs.